### ISSUE TITLE
moo-cli external templates

### ISSUE THREAD

**Issue by pablohpsilva** [SoftboxLab/moo-cli#44]
## Explanation
Hello everyone. I was wondering if I could implement something really cool on our project: A way to change `moo-cli` behaviour based on a configuration on the `package.json` file or a `.mooclirc` file.

## How
I believe I can make a few changes on `moo-cli` in order to make available a way to change the cli behaviour on creating components and flux code. I was thinking on creating a `.mooclirc` file and/or read the `package.json` from the user's project and check if there's any configuration specific to the cli. Examples:

`.mooclirc` file:
```json
{
    general: {
        framework: 'react',
        lint: '@softboxlab/eslint-config-gandalf-lint-react',
        typescript: true
    },
    component: {
        path: './src/weird/path',
        type: 'func',
        createTest: true
    },
    flux: {
        actionPath: './src/flux/actions',
        reducerPath: './src/flux/stores/reducers'
    }
}
``` 

`package.json` file
```
{
    ...
    moocli: {
        general: {
            framework: 'react',
            lint: '@softboxlab/eslint-config-gandalf-lint-react',
            typescript: true
        },
        component: {
            path: './src/weird/path',
            type: 'func',
            createTest: true
        },
        flux: {
            actionPath: './src/flux/actions',
            reducerPath: './src/flux/stores/reducers'
        }
    }
    ...
}
```

## Why
Doing so the user will have to type less, making the command line be as simple as: `moo <command> <name>`. No extra params needed. If the user needs something different than the default he set himself, then the command line will be the same. What is the advantage of using this compared to the CLI menu? Well... I believe I can save time on writing just the name of the thing I want instead of checking multiples "checkboxes".

**Comment by alexbruno** [comment]
Agree, but I will suggest some changes to the pattern.

1. `general.lint` is unecessary, gandalf will be present into **all** moo-cli generated projects, the "flavor" can be infered by `general.framework` option
2. I vote to simplify keys: `createTest => test` , `actionPath => actions` , `reducerPath => reducers` ...
3. these settings will only work inside an existing project, where already exists a `package.json`
4. `moo-cli arch <project>` command can generate these settings automaticaly during architecture bootstrap with default options

**Comment by pablohpsilva** [comment]
So do you think it will be a good idea to have this optional configs on the `package.json` instead having it on a separate file? Either way for me works, specially because the "default" way that `moo-cli` works will stay the same.

**Comment by pablohpsilva** [comment]
So your opinion was to change the object to this:
```javascript
{
    general: {
        framework: 'react',
        typescript: true
    },
    component: {
        path: './src/weird/path',
        type: 'func',
        test: true
    },
    flux: {
        action: './src/flux/actions',
        reducer: './src/flux/stores/reducers'
    },
    templates: {
        react: {
            components: {
                 comp: 'src/templates/react/components/comp.js',
                 func: 'src/templates/react/components/func.js'
            },
            ...
        }
    }
}
```

**Comment by alexbruno** [comment]
Yeah, I like the idea. Both `package.json` and `.rc` file are good to me.
The object above is OK, but use plurals to `actions` and `reducers`.
Remember these keys only makes sense for React projects. Maybe, for Vue projects, we can map a `modules` dir for Vuex store modules in the future.

```javascript
{
    ...
    flux: { // notice the 's'
        actions: './src/flux/actions',
        reducers: './src/flux/stores/reducers'
    }
    ...
}
```
